lib/core/providers/cart_provider.dart

```diff
@@ -0,0 +1,5 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:my_app/core/states/cart_state.dart';
+
+final cartProvider =
+    StateNotifierProvider<CartNotifier, CartState>((ref) => CartNotifier());
```

Comment:
As noted, you can use riverpod with either The ChangeNotifierProvider or StateNotifierProvider --> 
StateNotifierProvider is a little bit more complex, but gives you more freedoms